File: 0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int startIdx = 0;
        int totalTank = 0;
        int currTank = 0;
        for (int i = 0; i < gas.length; i++){
            int gain = gas[i] - cost[i];
            totalTank += gain;
            currTank += gain;
           if(currTank < 0){
            startIdx = i + 1;
            currTank = 0;
           }

        }
        return totalTank >= 0 ? startIdx : -1;
    }
}

File: 0134-gas-station/0134-gas-station.ts
function canCompleteCircuit(gas: number[], cost: number[]): number {
    let sum1 = 0
    let sum2 = 0
    for (let i = 0; i < gas.length; i++){
                sum1 = sum1 + gas[i]
                sum2 = sum2 + cost[i]                  
    }
    if (sum1<sum2){
        return -1
    }
    let tank = 0; 
    let start = 0;
    for (let i = 0; i < gas.length; i++){
        tank = tank + gas[i] - cost[i]
        if (tank < 0){
            start = i + 1
             tank = 0
        }
       
    }
    return start
}
